src/Dibi/Drivers/PdoDriver.php

```diff
@@ -89,31 +89,44 @@ public function disconnect(): void
 	 */
 	public function query(string $sql): ?Dibi\ResultDriver
 	{
-		$res = $this->connection->query($sql);
-		if ($res) {
-			$this->affectedRows = $res->rowCount();
-			return $res->columnCount() ? $this->createResultDriver($res) : null;
+		try {
+			$res = $this->connection->query($sql);
+
+			if ($res) {
+				$this->affectedRows = $res->rowCount();
+				return $res->columnCount() ? $this->createResultDriver($res) : null;
+			}
+
+		} catch (\PDOException $pdoException) {
+			$this->affectedRows = null;
+			throw $this->createException($pdoException->errorInfo, $sql, $pdoException);
```

Comment:
I do not like that I need to pass `$previous` all around. Any ideas how to do better?

Reply:
I think the PDOException does not contain any additional information, so there is no need to pass on the previous one.

Reply:
Hmm, as it is implementation detail, no code ever should depend on that. Gonna remove that